### ISSUE TITLE
Feature request: Check against subset of args emitted in events

### ISSUE THREAD

**Issue by alsco77** [TrueFiEng/Waffle#437]
## Context 
Sometimes it is very challenging to get the exact value you expect to be emitted from an event. Being able to pass a partial object would solve this problem by letting the listener know exactly what to expect.

Right now, you either need to specify all args, or none, which causes me to remove the args completely from a lot of the checks.

## Solution

Support passing of an object to the `.withArgs` fn with specific things to check

**Comment by marekkirejczyk** [comment]
Interesting idea. Would you like to make a PR?

**Comment by naddison36** [comment]
I've added `withNamedArgs` matcher to the Chai `emit` matcher in PR  https://github.com/EthWorks/Waffle/pull/456
This takes an object with name and value of the arguments to be checked in the event.

**Comment by paulrberg** [comment]
> Sometimes it is very challenging to get the exact value you expect to be emitted from an event.

Seconded. I just stumbled upon a situation whereby one of the event arguments depends on the current block timestamp, and since I don't know the exact timestamp until I execute the non-constant contract call, I can't calculate it beforehand.

**Comment by rzadp** [comment]
Just letting know that the issue is on our radar.

**Comment by paulrberg** [comment]
For anyone interested: the recently released [Hardhat v2.10.0](https://github.com/NomicFoundation/hardhat/releases/tag/hardhat%402.10.0) offers this feature (checking against subset of event args) as part of their Hardhat Chai Matchers:

> - Better matching of event arguments
>   - You can ignore some argument and only check the rest

**Comment by rzadp** [comment]
This is now available in Waffle 4.
https://ethereum-waffle.readthedocs.io/en/latest/matchers.html#emitting-events

e.g. :

```
await expect(token.transfer(walletTo.address, 7))
  .to.emit(token, 'Transfer')
  .withNamedArgs({
    from: wallet.address,
    to: walletTo.address,
    amount: 7
  });
```

Individual named arguments can be omitted.
